src/modules/CurrencyModule.h

```diff
@@ -1,23 +1,43 @@
 #pragma once
+#include <qtimer.h>
 #include "core/Module.h"
+#include <random>
 
-class CurrencyModule : public Module
-{
-public:
-	CurrencyModule();
+class UmikoBot;
 
-	void OnMessage(Discord::Client& client, const Discord::Message& message) override;
+#define randGiveawayProb 0.001
+
+class CurrencyModule : public Module {
+private:
+	struct Setting {
+		double currency;
+		bool isDailyClaimed;
+	};
+
+	QMap<snowflake_t, Setting> m_settings;
+	std::random_device random_device;
+	std::mt19937 random_engine{ random_device() };
+	std::bernoulli_distribution randp{ randGiveawayProb };
+	QTimer m_timer;
+	bool isRandomGiveAwayDone{ false };
+	bool allowGiveaway{ false };
+	snowflake_t giveawayChannelId{0};
+	QString currencyName;
+	QString currencySymbol;
 
-	void StatusCommand(QString& result, snowflake_t guild, snowflake_t user) override;
+	struct selfGamble {
```

Comment:
I'd appreciate if this was renamed to `SelfGamble` so it isn't confused with being a variable.

Reply:
Gotcha!